File: javascript/src/model/ForwardingNumberInfo.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.RingCentralApi) {
      root.RingCentralApi = {};
    }
    root.RingCentralApi.ForwardingNumberInfo = factory(root.RingCentralApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The ForwardingNumberInfo model module.
   * @module model/ForwardingNumberInfo
   * @version v1.0
   */

  /**
   * Constructs a new <code>ForwardingNumberInfo</code>.
   * @alias module:model/ForwardingNumberInfo
   * @class
   */
  var exports = function() {







  };

  /**
   * Constructs a <code>ForwardingNumberInfo</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/ForwardingNumberInfo} obj Optional instance to populate.
   * @return {module:model/ForwardingNumberInfo} The populated <code>ForwardingNumberInfo</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('id')) {
        obj['id'] = ApiClient.convertToType(data['id'], 'String');
      }
      if (data.hasOwnProperty('uri')) {
        obj['uri'] = ApiClient.convertToType(data['uri'], 'String');
      }
      if (data.hasOwnProperty('phoneNumber')) {
        obj['phoneNumber'] = ApiClient.convertToType(data['phoneNumber'], 'String');
      }
      if (data.hasOwnProperty('label')) {
        obj['label'] = ApiClient.convertToType(data['label'], 'String');
      }
      if (data.hasOwnProperty('features')) {
        obj['features'] = ApiClient.convertToType(data['features'], 'String');
      }
      if (data.hasOwnProperty('flipNumber')) {
        obj['flipNumber'] = ApiClient.convertToType(data['flipNumber'], 'Integer');
      }
    }
    return obj;
  }


  /**
   * Internal identifier of a forwarding/call flip phone number
   * @member {String} id
   */
  exports.prototype['id'] = undefined;

  /**
   * Canonical URI of a forwarding/call flip phone number
   * @member {String} uri
   */
  exports.prototype['uri'] = undefined;

  /**
   * Forwarding/Call flip phone number
   * @member {String} phoneNumber
   */
  exports.prototype['phoneNumber'] = undefined;

  /**
   * Forwarding/Call flip number title
   * @member {String} label
   */
  exports.prototype['label'] = undefined;

  /**
   * Type of option this phone number is used for. Multiple values are accepted
   * @member {module:model/ForwardingNumberInfo.FeaturesEnum} features
   */
  exports.prototype['features'] = undefined;

  /**
   * Number assigned to the call flip phone number, corresponds to the shortcut dial number
   * @member {Integer} flipNumber
   */
  exports.prototype['flipNumber'] = undefined;


  /**
   * Allowed values for the <code>features</code> property.
   * @enum {String}
   * @readonly
   */
  exports.FeaturesEnum = { 
    /**
     * value: CallFlip
     * @const
     */
    CALLFLIP: "CallFlip",
    
    /**
     * value: CallForwarding
     * @const
     */
    CALLFORWARDING: "CallForwarding"
  };

  return exports;
}));

File: csharp/src/main/csharp/IO/Swagger/Model/ForwardingNumberInfo.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace IO.Swagger.Model
{
    /// <summary>
    /// 
    /// </summary>
    [DataContract]
    public partial class ForwardingNumberInfo :  IEquatable<ForwardingNumberInfo>
    { 
    
        /// <summary>
        /// Type of option this phone number is used for. Multiple values are accepted
        /// </summary>
        /// <value>Type of option this phone number is used for. Multiple values are accepted</value>
        [JsonConverter(typeof(StringEnumConverter))]
        public enum FeaturesEnum {
            
            [EnumMember(Value = "CallFlip")]
            Callflip,
            
            [EnumMember(Value = "CallForwarding")]
            Callforwarding
        }

    
        /// <summary>
        /// Type of option this phone number is used for. Multiple values are accepted
        /// </summary>
        /// <value>Type of option this phone number is used for. Multiple values are accepted</value>
        [DataMember(Name="features", EmitDefaultValue=false)]
        public FeaturesEnum? Features { get; set; }
    
        /// <summary>
        /// Initializes a new instance of the <see cref="ForwardingNumberInfo" /> class.
        /// Initializes a new instance of the <see cref="ForwardingNumberInfo" />class.
        /// </summary>
        /// <param name="Id">Internal identifier of a forwarding/call flip phone number.</param>
        /// <param name="Uri">Canonical URI of a forwarding/call flip phone number.</param>
        /// <param name="PhoneNumber">Forwarding/Call flip phone number.</param>
        /// <param name="Label">Forwarding/Call flip number title.</param>
        /// <param name="Features">Type of option this phone number is used for. Multiple values are accepted.</param>
        /// <param name="FlipNumber">Number assigned to the call flip phone number, corresponds to the shortcut dial number.</param>

        public ForwardingNumberInfo(string Id = null, string Uri = null, string PhoneNumber = null, string Label = null, FeaturesEnum? Features = null, int? FlipNumber = null)
        {
            this.Id = Id;
            this.Uri = Uri;
            this.PhoneNumber = PhoneNumber;
            this.Label = Label;
            this.Features = Features;
            this.FlipNumber = FlipNumber;
            
        }
        
    
        /// <summary>
        /// Internal identifier of a forwarding/call flip phone number
        /// </summary>
        /// <value>Internal identifier of a forwarding/call flip phone number</value>
        [DataMember(Name="id", EmitDefaultValue=false)]
        public string Id { get; set; }
    
        /// <summary>
        /// Canonical URI of a forwarding/call flip phone number
        /// </summary>
        /// <value>Canonical URI of a forwarding/call flip phone number</value>
        [DataMember(Name="uri", EmitDefaultValue=false)]
        public string Uri { get; set; }
    
        /// <summary>
        /// Forwarding/Call flip phone number
        /// </summary>
        /// <value>Forwarding/Call flip phone number</value>
        [DataMember(Name="phoneNumber", EmitDefaultValue=false)]
        public string PhoneNumber { get; set; }
    
        /// <summary>
        /// Forwarding/Call flip number title
        /// </summary>
        /// <value>Forwarding/Call flip number title</value>
        [DataMember(Name="label", EmitDefaultValue=false)]
        public string Label { get; set; }
    
        /// <summary>
        /// Number assigned to the call flip phone number, corresponds to the shortcut dial number
        /// </summary>
        /// <value>Number assigned to the call flip phone number, corresponds to the shortcut dial number</value>
        [DataMember(Name="flipNumber", EmitDefaultValue=false)]
        public int? FlipNumber { get; set; }
    
        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class ForwardingNumberInfo {\n");
            sb.Append("  Id: ").Append(Id).Append("\n");
            sb.Append("  Uri: ").Append(Uri).Append("\n");
            sb.Append("  PhoneNumber: ").Append(PhoneNumber).Append("\n");
            sb.Append("  Label: ").Append(Label).Append("\n");
            sb.Append("  Features: ").Append(Features).Append("\n");
            sb.Append("  FlipNumber: ").Append(FlipNumber).Append("\n");
            
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="obj">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object obj)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            return this.Equals(obj as ForwardingNumberInfo);
        }

        /// <summary>
        /// Returns true if ForwardingNumberInfo instances are equal
        /// </summary>
        /// <param name="other">Instance of ForwardingNumberInfo to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(ForwardingNumberInfo other)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            if (other == null)
                return false;

            return 
                (
                    this.Id == other.Id ||
                    this.Id != null &&
                    this.Id.Equals(other.Id)
                ) && 
                (
                    this.Uri == other.Uri ||
                    this.Uri != null &&
                    this.Uri.Equals(other.Uri)
                ) && 
                (
                    this.PhoneNumber == other.PhoneNumber ||
                    this.PhoneNumber != null &&
                    this.PhoneNumber.Equals(other.PhoneNumber)
                ) && 
                (
                    this.Label == other.Label ||
                    this.Label != null &&
                    this.Label.Equals(other.Label)
                ) && 
                (
                    this.Features == other.Features ||
                    this.Features != null &&
                    this.Features.Equals(other.Features)
                ) && 
                (
                    this.FlipNumber == other.FlipNumber ||
                    this.FlipNumber != null &&
                    this.FlipNumber.Equals(other.FlipNumber)
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            // credit: http://stackoverflow.com/a/263416/677735
            unchecked // Overflow is fine, just wrap
            {
                int hash = 41;
                // Suitable nullity checks etc, of course :)
                
                if (this.Id != null)
                    hash = hash * 59 + this.Id.GetHashCode();
                
                if (this.Uri != null)
                    hash = hash * 59 + this.Uri.GetHashCode();
                
                if (this.PhoneNumber != null)
                    hash = hash * 59 + this.PhoneNumber.GetHashCode();
                
                if (this.Label != null)
                    hash = hash * 59 + this.Label.GetHashCode();
                
                if (this.Features != null)
                    hash = hash * 59 + this.Features.GetHashCode();
                
                if (this.FlipNumber != null)
                    hash = hash * 59 + this.FlipNumber.GetHashCode();
                
                return hash;
            }
        }

    }
}
